### DISCUSSION TITLE
DALL-E 3 on Azure OpenAI repeatedly returning same image.

### DISCUSSION THREAD

**Discussion by Uralstech** [RageAgainstThePixel/com.openai.unity#272] [Q&A]
# Bug Report

## Overview
I was trying DALL-E 3 on Azure OpenAI, and it seems there is a problem with the download caching.

The first time I run an image generation request it returns the image correctly. After that, any query returns that same image. When I clear the cache for my app, it generates a new image, then every subsequent query returns that.

When I call `Rest.DeleteDownloadCache();` before each query it works every time. Strangely, this issue is only there for Azure OpenAI requests and *not* for normal OpenAI DALL-E 3 requests.

This is part of the code I'm using:
```csharp
    public OpenAIClient OpenAIClientDallE;
    
    private void Start()
    {
        OpenAIClientDallE = new OpenAIClient(_azureDallEConfig);
    }

    private async Task<string> GenerateImage(string prompt)
    {
        Debug.Log($"Generating image: \"{prompt}\"");

        ImageGenerationRequest request = new(prompt, Model.DallE_3);

        // Rest.DeleteDownloadCache();
        IReadOnlyList<ImageResult> imageResults = await OpenAIClient.ImagesEndPoint.GenerateImageAsync(request);

        foreach (ImageResult result in imageResults)
        {
            RawImage image = Instantiate(_imagePrefab, _contentTransform);
            image.texture = result.Texture;
        }

        return "The image was generated successfully and has been displayed to the user.";
    }
```

The GenerateImage method is called by my GPT-4o powered assistant as a function call. I have only tried this on Android, on my S10 running Android 12. I'm using Unity 2022.3.38f1.

**Comment by StephenHodgson** [comment]
@Uralstech I just want to confirm, that when you use OpenAI you get a new image?

When you use Azure it is the same image from the previous request?

**Reply by StephenHodgson** [reply]
I'm also gonna test this in [OpenAI-DotNet](https://github.com/RageAgainstThePixel/OpenAI-DotNet) just to verify that it isn't an issue with the Unity package.

**Reply by Uralstech** [reply]
Yup. Thanks for the quick reply!

**Reply by StephenHodgson** [reply]
I'll check it out. thanks for the report!

**Reply by Uralstech** [reply]
You're welcome!
